### PR TITLE
DEV: refactor JS files to not use `self = this` in code.

### PR DESCRIPTION
**Pull request by vinothkannans**
We no longer use this pattern. Instead, we can use `bind(this)` to the functions.

### CODE REVIEW

File: app/assets/javascripts/discourse/app/lib/debounce.js [hunk 1]
```diff
@@ -5,15 +5,14 @@ import { debounce } from "@ember/runloop";
...
-    func.apply(self, args);
+    func.apply(this, args);
   };
 
   return function () {
-    self = this;
     args = arguments;
 
-    debounce(null, later, wait);
+    debounce(null, later.bind(this), wait);
```
Comment (vinothkannans) [inline app/assets/javascripts/discourse/app/lib/debounce.js:16]: I'm not sure about how to test this in the local dev environment 🤔

File: app/assets/javascripts/discourse/app/components/topic-list.js [hunk 2]
```diff
@@ -170,14 +170,13 @@ export default Component.extend(LoadMore, {
   },
 
   click(e) {
-    let self = this;
     let onClick = function (sel, callback) {
```
Comment (eviltrout) [inline app/assets/javascripts/discourse/app/components/topic-list.js:173]: You don't need a `bind(this)` if you use the fat arrow `(sel, callback) =>` syntax which we prefer.

File: app/assets/javascripts/discourse/app/lib/debounce.js (hunk 1 above)
Comment (eviltrout) [reply app/assets/javascripts/discourse/app/lib/debounce.js:16]: The same `=>` note from above should do it.

**Comment by SamSaffron** [comment]
@vinothkannans can you update this PR

pushing to 2.9, does not seem urgent.

**Comment by SamSaffron** [comment]
@vinothkannans can you handle @eviltrout's feedback?

**Comment by vinothkannans** [comment]
sorry for the delay. this is done @eviltrout

**Review by eviltrout — APPROVED** [review]

**Pull request merged**